erpnext/assets/doctype/location/location.py

```diff
@@ -223,3 +223,10 @@ def add_node():
 
 def on_doctype_update():
 	frappe.db.add_index("Location", ["lft", "rgt"])
+
+@frappe.whitelist()
+def get_total_location():
+	# Get the total of all locations in square meters
+	total_location = frappe.db.get_value("Location", {"is_group": 0}, "sum(round(area, 3))")
```

Comment:
Open the tree view for location and select a location in the filter above. Notice that the tree view will change and the selected filter will become the root of the structure but its area would be wrongly calculated.

Reply:
I'm sorry, it's true. I solved it:

![solved_tree_view_location](https://user-images.githubusercontent.com/26640728/47382915-ada2e980-d6c0-11e8-872d-1c1f96aa33cc.gif)

Reply:
The suggested changes have been applied to the following  5705af8